[2023-03-16T05:12:08.551+0000] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: Spotify_Data_ETL.ValidateAndLoadArtistDim.ValidateArtistDim manual__2023-03-16T05:02:48.467376+00:00 [queued]>
[2023-03-16T05:12:08.559+0000] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: Spotify_Data_ETL.ValidateAndLoadArtistDim.ValidateArtistDim manual__2023-03-16T05:02:48.467376+00:00 [queued]>
[2023-03-16T05:12:08.559+0000] {taskinstance.py:1279} INFO - 
--------------------------------------------------------------------------------
[2023-03-16T05:12:08.561+0000] {taskinstance.py:1280} INFO - Starting attempt 10 of 12
[2023-03-16T05:12:08.561+0000] {taskinstance.py:1281} INFO - 
--------------------------------------------------------------------------------
[2023-03-16T05:12:08.579+0000] {taskinstance.py:1300} INFO - Executing <Task(PythonOperator): ValidateAndLoadArtistDim.ValidateArtistDim> on 2023-03-16 05:02:48.467376+00:00
[2023-03-16T05:12:08.585+0000] {standard_task_runner.py:55} INFO - Started process 9333 to run task
[2023-03-16T05:12:08.594+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'Spotify_Data_ETL', 'ValidateAndLoadArtistDim.ValidateArtistDim', 'manual__2023-03-16T05:02:48.467376+00:00', '--job-id', '1220', '--raw', '--subdir', 'DAGS_FOLDER/spotifydag.py', '--cfg-path', '/tmp/tmpt1tp_ytd']
[2023-03-16T05:12:08.595+0000] {standard_task_runner.py:83} INFO - Job 1220: Subtask ValidateAndLoadArtistDim.ValidateArtistDim
[2023-03-16T05:12:08.743+0000] {task_command.py:388} INFO - Running <TaskInstance: Spotify_Data_ETL.ValidateAndLoadArtistDim.ValidateArtistDim manual__2023-03-16T05:02:48.467376+00:00 [running]> on host 18c42e4ef76b
[2023-03-16T05:12:09.237+0000] {taskinstance.py:1509} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=Ananta Moharana
AIRFLOW_CTX_DAG_ID=Spotify_Data_ETL
AIRFLOW_CTX_TASK_ID=ValidateAndLoadArtistDim.ValidateArtistDim
AIRFLOW_CTX_EXECUTION_DATE=2023-03-16T05:02:48.467376+00:00
AIRFLOW_CTX_TRY_NUMBER=10
AIRFLOW_CTX_DAG_RUN_ID=manual__2023-03-16T05:02:48.467376+00:00
[2023-03-16T05:12:09.239+0000] {logging_mixin.py:137} INFO - None
[2023-03-16T05:12:09.239+0000] {logging_mixin.py:137} INFO - ['artist_name', 'id']
[2023-03-16T05:12:09.240+0000] {logging_mixin.py:137} INFO - artist_dimension
[2023-03-16T05:12:09.243+0000] {taskinstance.py:1768} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 175, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 192, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/src/transform.py", line 74, in validate_data
    data_table=pd.read_json(data)
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/util/_decorators.py", line 207, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/util/_decorators.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/json/_json.py", line 614, in read_json
    return json_reader.read()
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/json/_json.py", line 748, in read
    obj = self._get_object_parser(self.data)
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/json/_json.py", line 770, in _get_object_parser
    obj = FrameParser(json, **kwargs).parse()
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/json/_json.py", line 885, in parse
    self._parse_no_numpy()
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/json/_json.py", line 1140, in _parse_no_numpy
    loads(json, precise_float=self.precise_float), dtype=None
ValueError: Unexpected character found when decoding 'NaN'
[2023-03-16T05:12:09.261+0000] {taskinstance.py:1323} INFO - Marking task as UP_FOR_RETRY. dag_id=Spotify_Data_ETL, task_id=ValidateAndLoadArtistDim.ValidateArtistDim, execution_date=20230316T050248, start_date=20230316T051208, end_date=20230316T051209
[2023-03-16T05:12:09.275+0000] {standard_task_runner.py:105} ERROR - Failed to execute job 1220 for task ValidateAndLoadArtistDim.ValidateArtistDim (Unexpected character found when decoding 'NaN'; 9333)
[2023-03-16T05:12:09.345+0000] {local_task_job.py:208} INFO - Task exited with return code 1
[2023-03-16T05:12:09.397+0000] {taskinstance.py:2578} INFO - 0 downstream tasks scheduled from follow-on schedule check
